test/src/xtdb/fixtures/checkpoint_store.clj

```diff
@@ -52,3 +67,41 @@
 
           (t/is (= "Hey Ivan!"
                    (slurp (io/file dest-dir "ivan.txt")))))))))
+
+(defn- sync-path-throw [^Path from-root-path ^Path to-root-path]
+  (doseq [^Path from-path (-> (Files/walk from-root-path Integer/MAX_VALUE (make-array FileVisitOption 0))
+                              .iterator
+                              iterator-seq)
+          :let [to-path (.resolve to-root-path (str (.relativize from-root-path from-path)))]]
+    (cond
+      (Files/isDirectory from-path (make-array LinkOption 0))
+      (Files/createDirectories to-path (make-array FileAttribute 0))
+
+      (Files/isRegularFile from-path (make-array LinkOption 0))
+      (Files/copy from-path to-path ^"[Ljava.nio.file.CopyOption;" (make-array CopyOption 0))))
+  (throw (Exception. "broken!")))
+
+(defn test-checkpoint-broken-store-failed-download
```

Comment:
`deftest`?

Reply:
well the test is defined in `xtdb/checkpoint-test` (test-fs-checkpoint-store-failed-download).
Just being consistent with an existing test defined above that does just that.